nexus_common/include/nexus_common/task_remapper.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef NEXUS_COMMON__TASK_REMAPPER_HPP
+#define NEXUS_COMMON__TASK_REMAPPER_HPP
+
+#include "nexus_common_export.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+
+namespace nexus::common {
+
+/**
+ * Provides task remapping capability
+ */
+class NEXUS_COMMON_EXPORT TaskRemapper
+{
+public:
+  /*
+   * Initialize the remapper with the value of a ROS parameter containing a YAML
+   */
+  TaskRemapper(const std::string& param);
+
+  /*
+   * Remaps, if necessary, the input task
+   */
+  std::string remap(const std::string& task) const;
```

Comment:
The current implementation returns `task` if no remapping is specified which 1) is an implicit behavior and 2) could give API users a false sense that a remapping was performed. Suggest to have the API return an `std::optional<std::string>` instead of no remapping was specified and handle the optional case in the orchestrator. Yes it adds some duplication in the orchestration logic but the API is clearer and users could potentially save one unnecessary copy of the string here if no remapping is found.

Reply:
https://github.com/osrf/nexus/pull/55/commits/b881d1e2046602ce3f909ac9df26412bc5935e94